aws-lc-rs/src/ed25519.rs

```diff
@@ -116,31 +117,22 @@ impl KeyPair for Ed25519KeyPair {
     }
 }
 
-pub(crate) unsafe fn generate_key(rng: &dyn SecureRandom) -> Result<LcPtr<EVP_PKEY>, ()> {
-    // TODO: Should we drop support for using the provided rng in a minor release?
-
-    let mut seed = [0u8; ED25519_SEED_LEN];
-    rng.fill(&mut seed)?;
-
-    let mut public_key = MaybeUninit::<[u8; ED25519_PUBLIC_KEY_LEN]>::uninit();
-    let mut private_key = MaybeUninit::<[u8; ED25519_PRIVATE_KEY_LEN]>::uninit();
-    ED25519_keypair_from_seed(
-        public_key.as_mut_ptr().cast(),
-        private_key.as_mut_ptr().cast(),
-        seed.as_ptr(),
-    );
-    seed.zeroize();
-
-    // ED25519_keypair_from_seed doesn't set FIPS indicator, and Ed25119 is not approved anyways at this time.
-    // Seems like it could be approved for use in the future per FIPS 186-5 and CMVP guidance.
-    crate::fips::set_fips_service_status_unapproved();
-
-    LcPtr::new(EVP_PKEY_new_raw_private_key(
-        EVP_PKEY_ED25519,
-        null_mut(),
-        private_key.assume_init().as_ptr(),
-        ED25519_PRIVATE_KEY_SEED_LEN,
-    ))
+pub(crate) unsafe fn generate_key() -> Result<LcPtr<EVP_PKEY>, ()> {
+    let pkey_ctx = LcPtr::new(unsafe { EVP_PKEY_CTX_new_id(EVP_PKEY_ED25519, null_mut()) })?;
+
+    if 1 != unsafe { EVP_PKEY_keygen_init(*pkey_ctx) } {
+        return Err(());
+    }
+
+    let mut pkey = null_mut::<EVP_PKEY>();
+
+    if 1 != indicator_check!(unsafe { EVP_PKEY_keygen(*pkey_ctx, &mut pkey) }) {
```

Comment:
Not expected to set the indicator, but since `EVP_PKEY_keygen` would set the indicator if it were approved, it seemed like it made sense to just put it here. That way I can have a test that validates that this is not an approved API explicitly.

Where before, I was just manually toggling our internal indicator state in the code for removal.